scripts/core/editor3/components/spellchecker/SpellcheckerContextMenu.tsx

```diff
@@ -17,46 +21,71 @@ export class SpellcheckerContextMenuComponent extends React.Component<IProps> {
     dropdownElement: any;
 
     componentDidMount() {
-        this.stickyElementTracker = new StickElementsWithTracking(this.props.targetElement, this.dropdownElement);
+        this.stickyElementTracker = new StickElementsWithTracking(
+            this.props.targetElement,
+            this.dropdownElement,
+        );
     }
     componentWillUnmount() {
         this.stickyElementTracker.destroy();
     }
 
+    onSuggestionClick(suggestion: ISpellcheckerSuggestion) {
+        this.props.dispatch(
+            actions.replaceWord({
+                word: {
+                    text: this.props.warning.text,
+                    offset: this.props.warning.startOffset,
+                },
+                newWord: suggestion.text,
+            }),
+        );
+    }
+
     render() {
-        const {suggestions} = this.props.warning;
-        const {spellchecker} = this.props;
+        const { suggestions, message } = this.props.warning;
+        const { spellchecker } = this.props;
+
+        // If the message exists, use it as the button text
+        // instead of the suggestion
+        const messageExists = Boolean(message);
 
         return (
             <div className={'dropdown open suggestions-dropdown'}
                 ref={(el) => this.dropdownElement = el}
                 style={{zIndex: 999, border: 'solid transparent', borderWidth: '6px 0'}}
                 data-test-id="spellchecker-menu"
             >
-                <ul className={'dropdown__menu'} style={{position: 'static'}}>
-                    <div className="form-label" style={{margin: '0 16px'}}>{gettext('Suggestions')}</div>
-                    {
-                        suggestions.length === 0
-                            ? <li><button>{gettext('SORRY, NO SUGGESTIONS.')}</button></li>
-                            : suggestions.map((suggestion, index) =>
+                <ul className={'dropdown__menu'} style={{ position: 'static' }}>
+                    <div className='form-label' style={{ margin: '0 16px' }}>
+                        {gettext('Suggestions')}
+                    </div>
+                    {messageExists && (
+                        <li>
+                            <button onMouseDown={() => this.onSuggestionClick(suggestions[0])}>{message}</button>
+                        </li>
+                    )}
+                    {!messageExists &&
```

Comment:
Are you using boolean logic to simulate ternaries? :D why?

Reply:
See comment above